gh-127371 Avoid unbounded growth SpooledTempfile.writelines (#127372)
python/cpython: Misc/NEWS.d/next/Security/2024-11-28-20-29-21.gh-issue-127371.PeEhUd.rst

> Avoid unbounded buffering for :meth:`!tempfile.SpooledTemporaryFile.writelines`. Previously, disk spillover was only checked after the lines iterator had been exhausted. This is now done after each line is written.

Reviewer: ```suggestion
Avoid unbounded buffering for :meth:`!tempfile.SpooledTemporaryFile.writelines`.
Previously, disk spillover was only checked after the lines iterator had been
exhausted. This is now done after each line is written.
```

> Avoid unbounded buffering for ``SpooledTemporary`` file. Previously, when the ``writelines`` method was called, disk-spillover was only checked after the lines iterator had been exhausted. This is now done while writing instead.

Reviewer 2: ```suggestion
Avoid unbounded buffering for :class:`tempfile.SpooledTemporaryFile`. Previously, when the :meth:`tempfile.SpooledTemporaryFile.writelines` method was called, disk-spillover was only checked after the lines iterator had been exhausted. This is now done while writing instead.
```

> Avoid unbounded buffering for :class:`tempfile.SpooledTemporaryFile`. Previously, when the :meth:`tempfile.SpooledTemporaryFile.writelines` method was called, disk-spillover was only checked after the lines iterator had been exhausted. This is now done while writing instead.

Reviewer 2: I think this is slightly more concise:
```suggestion
Avoid unbounded buffering for :meth:`tempfile.SpooledTemporaryFile.writelines`. Previously, disk spillover was only checked after the lines iterator had been exhausted. This is now done after each line is written.
```
